server.go

```diff
@@ -291,15 +291,30 @@ func (s *Server) handlerRoot(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	activeMovies, err := s.data.GetActiveMovies()
-	if err != nil {
-		s.doError(
-			http.StatusBadRequest,
-			fmt.Sprintf("Cannot get active movies: %v", err),
-			w, r)
-		return
-	}
+	movieList := []*common.Movie{}
 
+	if r.Body != http.NoBody {
+		err := r.ParseForm()
+		if err != nil {
+			fmt.Printf("[ERR] %v\n", err)
+		}
+		searchVal := r.FormValue("search")
+
+		movieList, err = s.data.SearchMovieTitles(searchVal)
+		if err != nil {
+			fmt.Printf("[ERR] %v\n", err)
+		}
+	} else {
+		var err error = nil
+		movieList, err = s.data.GetActiveMovies()
+		if err != nil {
+			s.doError(
+				http.StatusBadRequest,
+				fmt.Sprintf("Cannot get active movies: %v", err),
```

Comment:
This shouldn't send the real error message to the client.  That should only be printed for Admins and to the console because it is possible to contain database server details that should not be public.

Reply:
fixed with 55f9201